js/form.js

```diff
@@ -1,114 +1,43 @@
-};
-//Активация формы
-const formActivate = () => {
-  mapFiltersFeatures.disabled = false;
-
-  for(let i=0; i<adFormElement.length; i++) {
-    adFormElement[i].disabled = false;
-  }
-
-  for(let i=0; i<mapFiltersSelect.length; i++) {
-    mapFiltersSelect[i].disabled = false;
-  }
-
-  adForm.classList.remove('ad-form--disabled');
-  mapFilters.classList.remove('ad-form--disabled');
-};
-//Валидация заголовка
-const cardNameValidate = cardNameInput.addEventListener('input', () => {
-  const valueLength = cardNameInput.value.length;
-
-  if (valueLength < MIN_NAME_LENGTH) {
-    cardNameInput.setCustomValidity(`Ещё ${  MIN_NAME_LENGTH - valueLength } симв.`);
-  } else if (valueLength > MAX_NAME_LENGTH) {
-    cardNameInput.setCustomValidity(`Удалите лишние ${  valueLength - MAX_NAME_LENGTH } симв.`);
-  } else {
-    cardNameInput.setCustomValidity('');
-  }
-
-  cardNameInput.reportValidity();
-});
 //Подмена минимального значения цены за жилье в завичисимости от типа
-const addMinPrice = () => {
-  const typeValue = typeHousesSelect.value;
-  const priceMin = PRICE_MIN[typeValue];
-  cardPrice.placeholder = priceMin;
-  cardPrice.min = priceMin;
+const validateMinPrice = () => {
+  addForm.price.min = TYPES_OF_HOUSING[addForm.type.value].minPrice;
+  addForm.price.placeholder = TYPES_OF_HOUSING[addForm.type.value].minPrice;
 };
 
 //Валидация количества гостей и комнат
-const validateRoomsAndGuests = () => {
-  switch (roomNumberSelect.value) {
-    case '1':
-      if (capacitySelect.value === '3' || capacitySelect.value === '2' || capacitySelect.value === '0') {
-        roomNumberSelect.setCustomValidity('Только для 1 гостя');
-      } else {
-        roomNumberSelect.setCustomValidity('');
-      }
-      break;
-
-    case '2':
-      if (capacitySelect.value === '3' || capacitySelect.value === '0') {
-        roomNumberSelect.setCustomValidity('Только для 1-го или 2-х гостей');
-      } else {
-        roomNumberSelect.setCustomValidity('');
-      }
-      break;
-
-    case '3':
-      if (capacitySelect.value === '0') {
-        roomNumberSelect.setCustomValidity('Только для 1-го, 2-х или 3-х гостей');
-      } else {
-        roomNumberSelect.setCustomValidity('');
-      }
-      break;
-
-    case '100':
-      if (capacitySelect.value !== '0') {
-        roomNumberSelect.setCustomValidity('Не для гостей');
-      } else {
-        roomNumberSelect.setCustomValidity('');
-      }
-      break;
-
-    default:
-      roomNumberSelect.setCustomValidity('');
-      break;
+const validateRoomsInput = (evt) => {
+  const capacityList = ROOMS_FOR_GUESTS_MAP[evt.target.value];
+  const capacityElement = addForm.capacity.children;
+  for(let i=0; i<capacityElement.length; i++) {
+    if(capacityList.includes(capacityElement[i].value)) {
+      capacityElement[i].disabled = false;
+    } else {
+      capacityElement[i].disabled = true;
+    }
   }
+  addForm.capacity.value = capacityElement[0];
+};
```

Comment:
1. В каждой ветке if мы обращаемся к **capacityElement[i].disabled**. Получается дублирование, если перенести условие за знак равенства, то получится тот же результат, но более лаконичный: 
`capacityElement[i].disabled = !capacityList.includes(capacityElement[i].value)`. 

2. Запись можно сделать еще более читаемой, если использовать цикл [for...of](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Statements/for...of)

```suggestion
const validateRoomsInput = (evt) => {
  const capacityList = ROOMS_FOR_GUESTS_MAP[evt.target.value];
  for (let currentCapacityItem of adForm.capacity.children) {
    currentCapacityItem.disabled = !capacityList.includes(currentCapacityItem.value);
  };
  adForm.capacity.value = capacityList[0]; // тут нужно выбирать первый элемент из возможных иначе выбор работает некорректно, скриншот ниже
};
```
![image](https://user-images.githubusercontent.com/30810655/123546257-7e43b500-d764-11eb-83c7-78c12dbce5de.png)